lib/admin/WP_Auth0_Admin_Advanced.php

```diff
@@ -134,294 +134,233 @@ public function render_passwordless_method() {
-      </div>
-    <?php
+  public function render_passwordless_enabled( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Turns on passwordless login (email or SMS) in the Auth0 form. ', 'wp-auth0' ) .
+      __( 'Passwordless connections are managed in the ', 'wp-auth0' ) .
+      $this->get_dashboard_link( 'connections/passwordless' ) .
+      __( ' and at least one connection must be active for this to work. ', 'wp-auth0' ) .
+      __( 'Username and password login are not enabled when this is on', 'wp-auth0' )
+    );
   }
 
-  public function render_force_https_callback() {
-    $v = $this->options->get( 'force_https_callback' );
-
-    echo $this->render_a0_switch( "wpa0_force_https_callback", "force_https_callback", 1, 1 == $v );
-?>
+  public function render_force_https_callback( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'Forces the plugin to use HTTPS for the callback URL when a site supports both; ', 'wp-auth0' ) .
+      __( 'if disabled, the protocol from the WordPress home URL will be used', 'wp-auth0' )
+    );
+  }
 
-      <div class="subelement">
-        <span class="description"><?php echo __( 'This option forces the plugin to use HTTPS for the callback URL in those cases where it needs to support mixed HTTP and HTTPS pages. If disabled, it will pick the protocol from the WordPress home URL (configured under Settings > General).', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
+  public function render_remember_users_session( $args ) {
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
+    $this->render_field_description(
+      __( 'A user session by default is kept for two days. ', 'wp-auth0' ) .
+      __( 'Enabling this setting will extend that and make the session be kept for 14 days', 'wp-auth0' )
+    );
   }
 
-  public function render_remember_users_session() {
-    $v = $this->options->get( 'remember_users_session' );
+  public function render_migration_ws( $args ) {
+    $value = $this->options->get( $args[ 'opt_name' ] );
+    $this->render_switch( $args[ 'label_for' ], $args[ 'opt_name' ] );
 
-    echo $this->render_a0_switch( "wpa0_remember_users_session", "remember_users_session", 1, 1 == $v );
-?>
+    if ( $value ) {
 
-      <div class="subelement">
-        <span class="description"><?php echo __( 'Users session by default lives for two days. Enabling this setting will make the sessions be kept for 14 days.', 'wp-auth0' ); ?></span>
-      </div>
-    <?php
-  }
+      $this->render_field_description(
+        __( 'Users migration is enabled. ', 'wp-auth0' ) .
+        __( 'If you disable this setting, it must be re-enabled manually in the ', 'wp-auth0' ) .
```

Comment:
Is there a reason to have this option at all? Can't both enabling and disabling actions be performed directly on the dashboard, to prevent user error? Also think in that case if removing this would be considered breaking.

Reply:
Long conversation but yes, you are correct about how it _should_ be. This is something that needs to be dealt with carefully: 

https://wordpress.org/support/topic/dont-need-user-migration/